hathor/cli/side_dag.py

```diff
@@ -205,104 +195,82 @@ def is_option(arg_: str) -> bool:
-    hathor_node_process.start()
-    return hathor_node_process
+    args = _RunNodeArgs(
+        argv=argv,
+        runner=runner,
+        logging_output=logging_output,
+        capture_stdout=capture_stdout,
+        name=name
+    )
+    process = Process(target=_run_node, args=(args,), name=name)
+    process.start()
+    return process
 
 
-def _run_hathor_node(
-    argv: list[str],
-    run_node_cmd: type[RunNode],
-    logging_output: LoggingOutput,
-    capture_stdout: bool,
-    conn: 'Connection',
-) -> None:
-    """Function to be called by a background process to run the Hathor full node."""
-    from hathor.cli.util import process_logging_options, setup_logging
+@dataclass(frozen=True, slots=True, kw_only=True)
+class _RunNodeArgs:
+    argv: list[str]
+    runner: type[RunNode]
+    logging_output: LoggingOutput
+    capture_stdout: bool
+    name: str
+
 
-    # We don't terminate via SIGINT directly, instead the side-dag process will terminate us.
-    signal.signal(signal.SIGINT, lambda _, __: None)
+def _run_node(args: _RunNodeArgs) -> None:
+    from hathor.cli.util import process_logging_options, setup_logging
     try:
-        log_options = process_logging_options(argv)
+        log_options = process_logging_options(args.argv)
         setup_logging(
-            logging_output=logging_output,
+            logging_output=args.logging_output,
             logging_options=log_options,
-            capture_stdout=capture_stdout,
-            extra_log_info=_get_extra_log_info('hathor')
+            capture_stdout=args.capture_stdout,
+            extra_log_info=_get_extra_log_info(args.name)
         )
-        hathor_node = run_node_cmd(argv=argv)
-    except (BaseException, Exception):
-        logger.exception('hathor process terminated due to exception in initialization')
-        conn.send(HathorProcessInitFail(traceback.format_exc()))
+        logger.info(f'initializing node "{args.name}"')
+        node = args.runner(argv=args.argv)
+    except KeyboardInterrupt:
+        logger.warn(f'{args.name} node interrupted by user')
         return
-
-    conn.send(HathorProcessInitSuccess())
-    hathor_node.run()
-
-    # If `run()` returns, either the side-dag node exited and terminated us, leaving the message below, or the hathor
-    # node exited and we will terminate the side-dag node.
-    if conn.poll():
-        message = conn.recv()
-        assert isinstance(message, SideDagProcessTerminated)
-        logger.critical('hathor node terminated because side-dag process was terminated')
+    except (BaseException, Exception):
+        logger.exception(f'process "{args.name}" terminated due to exception in initialization')
         return
 
-    conn.send(HathorProcessTerminated())
-    logger.critical('terminating side-dag node...')
-    parent_pid = os.getppid()
-    parent_process = psutil.Process(parent_pid)
-    parent_process.terminate()
+    node.run()
+    logger.critical(f'node "{args.name}" terminated')
```

Comment:
Maybe `gracefully terminated`?

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/1079/commits/e965a13f46986b636960070abbf4693bea9fc5c7

Reply:
Ok! Thanks!